test/e2e/route_advertisements.go

```diff
@@ -463,3 +472,397 @@ spec:
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					statusPodA, err := userDefinedNetworkStatus(podsNetA[0], namespacedName(podsNetA[0].Namespace, cudnATemplate.Name))
+					framework.ExpectNoError(err)
+					// FIXME: L3 - curl returns code 7, the request works partially
+					return podsNetA[0].Spec.NodeName, "", net.JoinHostPort(statusPodA.IPs[ipIndex].IP.String(), "8080") + "/clientip", "", true
+				}),
+			ginkgo.Entry("host to an external UDN pod should not work",
+				func(ipIndex int) (clientName string, clientNamespace string, dst string, expectedOutput string, expectErr bool) {
+					// podsNetA[0] and podsNetA[2] are on different nodes
+					statusPodA, err := userDefinedNetworkStatus(podsNetA[0], namespacedName(podsNetA[0].Namespace, cudnATemplate.Name))
+					framework.ExpectNoError(err)
+					// FIXME: L3 - host to UDN pod on different node should not work
+					return podsNetA[2].Spec.NodeName, "", net.JoinHostPort(statusPodA.IPs[ipIndex].IP.String(), "8080") + "/clientip", "", false
+				}),
+		)
+
+	},
+	ginkgo.Entry("Layer3",
+		&udnv1.ClusterUserDefinedNetwork{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "bgp-udn-layer3-network-a",
+				Labels: map[string]string{"bgp-udn-layer3-network-a": ""},
+			},
+			Spec: udnv1.ClusterUserDefinedNetworkSpec{
+				Network: udnv1.NetworkSpec{
+					Topology: udnv1.NetworkTopologyLayer3,
+					Layer3: &udnv1.Layer3Config{
+						Role: "Primary",
+						Subnets: generateL3Subnets(udnv1.Layer3Subnet{
+							CIDR:       "102.102.0.0/16",
+							HostSubnet: 24,
+						}, udnv1.Layer3Subnet{
+							CIDR:       "2013:100:200::0/60",
+							HostSubnet: 64,
+						}),
+					},
+				},
+			},
+		}, &udnv1.ClusterUserDefinedNetwork{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   "bgp-udn-layer3-network-b",
+				Labels: map[string]string{"bgp-udn-layer3-network-b": ""},
+			},
+			Spec: udnv1.ClusterUserDefinedNetworkSpec{
+				Network: udnv1.NetworkSpec{
+					Topology: udnv1.NetworkTopologyLayer3,
+					Layer3: &udnv1.Layer3Config{
+						Role: "Primary",
+						Subnets: generateL3Subnets(udnv1.Layer3Subnet{
+							CIDR:       "103.103.0.0/16",
+							HostSubnet: 24,
+						}, udnv1.Layer3Subnet{
+							CIDR:       "2014:100:200::0/60",
+							HostSubnet: 64,
+						}),
+					},
+				},
+			},
+		},
+	),
+)
+
+func generateL3Subnets(v4, v6 udnv1.Layer3Subnet) []udnv1.Layer3Subnet {
+	var subnets []udnv1.Layer3Subnet
+	if isIPv4Supported() {
+		subnets = append(subnets, v4)
+	}
+	if isIPv6Supported() {
+		subnets = append(subnets, v6)
+	}
+	return subnets
+}
+
+// checkConnectivity performs a curl command from a specified client (pod or node)
+// to targetAddress. If clientNamespace is empty the function assumes clientName is a node that will be used as the
+// client.
+func checkConnectivity(clientName, clientNamespace, targetAddress string) (string, error) {
```

Comment:
nit: to have this within the describe as local func but dunno if that's possible with all the ginkgo entry stuff
check Connectivity is so generic a name but this function is redefined in other tests as local functions 
this is the first global function but it still should be reusable I guess